Library/Homebrew/sandbox.rb

```diff
@@ -33,6 +23,23 @@ def allow_write(path, options={})
                       :filter => filter
   end
 
+  def allow_write_temp_and_cache
+    allow_write "/private/tmp", :type => :subpath
+    allow_write "/private/var/folders", :type => :subpath
+    allow_write HOMEBREW_TEMP, :type => :subpath
+    allow_write HOMEBREW_CACHE, :type => :subpath
+  end
+
+  def allow_write_cellar(formula)
+    allow_write formula.rack, :type => :subpath
+    allow_write formula.etc, :type => :subpath
+    allow_write formula.var, :type => :subpath
+  end
+
+  def allow_write_log(formula)
+    allow_write HOMEBREW_LOGS/formula.name, :type => :subpath
+  end
```

Comment:
A few things here:

* So far every usage of the sandbox seems to imply `allow_write_temp_and_cache`. Maybe that should just be the default?

* Since everything seems to require `:type => :subpath`, it seems that API flexibility may be overkill? If you really anticipate we will need other types, maybe add an `allow_write_path` method rather than pass hashes around?